File: number_theory/array_pair_sum_divisibility_problem.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends
class Solution {
public:
  bool canPair(vector<int> nums, int k) {
    // Code here.
    if (nums.size() % 2 == 1)
      return false;
    int t = 0;
    vector<int> r(k);
    for (int i = 0; i < nums.size(); ++i) {
      r[nums[i] % k]++;
    }
    for (int i = 1; i < k; ++i) {
      if (r[i] != r[k - i])
        return false;
    }
    return r[0] % 2 == 0;
  }
};

//{ Driver Code Starts.
int main() {
  int tc;
  cin >> tc;
  while (tc--) {
    int n, k;
    cin >> n >> k;
    vector<int> nums(n);
    for (int i = 0; i < nums.size(); i++)
      cin >> nums[i];
    Solution ob;
    bool ans = ob.canPair(nums, k);
    if (ans)
      cout << "True\n";
    else
      cout << "False\n";
  }
  return 0;
}
// } Driver Code Ends

File: number_theory/array_pair_sum_divisibility_problem.py
# User function Template for python3


class Solution:
    def canPair(self, a, k):
        # Code here
        cnt = [0] * k
        for b in a:
            cnt[b % k] += 1

        if cnt[0] % 2 == 1 or k % 2 == 0 and cnt[k // 2] % 2 == 1:
            return False

        for i in range(1, k // 2):
            if cnt[i] != cnt[k - i]:
                return False

        return True


# {
# Driver Code Starts
# Initial Template for Python 3

if __name__ == "__main__":
    T = int(input())
    for i in range(T):
        n, k = input().split()
        n = int(n)
        k = int(k)
        nums = list(map(int, input().split()))
        ob = Solution()
        ans = ob.canPair(nums, k)
        if ans:
            print("True")
        else:
            print("False")
# } Driver Code Ends
